src/Database/Driver/sparql/StatementStub.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace Drupal\rdf_entity\Database\Driver\sparql;
+
+use Drupal\Core\Database\StatementInterface;
+
+/**
+ * Represents a faked database statement object.
+ *
+ * The Drupal core database logger cannot be swapped because, instead of being
+ * injected, is hardcoded in \Drupal\Core\Database\Database::startLog(). But the
+ * \Drupal\Core\Database\Log::log() is expecting a database statement of type
+ * \Drupal\Core\Database\StatementInterface as first argument and the SPARQL
+ * database driver uses no StatementInterface class. We workaround this
+ * limitation by faking a database statement object just to honour the logger
+ * requirement. We use a statement stub that only stores the connection and the
+ * query to be used when logging the event.
```

Comment:
Very clear explanation, thanks. I see that `Log::startLog()` is even a `final public static function`, it seems like core _really_ doesn't want it to be overridden. The reason why is explained in the class documentation for the `Log` class:

```
/**
 * Database query logger.
 *
 * We log queries in a separate object rather than in the connection object
 * because we want to be able to see all queries sent to a given database, not
 * database target. If we logged the queries in each connection object we
 * would not be able to track what queries went to which target.
 *
 * Every connection has one and only one logging object on it for all targets
 * and logging keys.
 */
```

This is a pity because `StatementInterface` is closely bound to PDO, and is not very useful for databases that do not have a PDO driver such as Virtuoso.